sql/core/src/main/scala/org/apache/spark/sql/internal/CatalogImpl.scala

```diff
@@ -338,7 +371,8 @@ class CatalogImpl(sparkSession: SparkSession) extends Catalog {
       tableType = tableType,
       storage = storage,
       schema = schema,
-      provider = Some(source)
+      provider = Some(source),
+      comment = Some(description)
```

Comment:
Also, it's a slight behaviour change I guess. If there's no description set, I think we should set `None` instead of `Some("")`.

Reply:
Good point. Fixed. I made `description` an `Option[String]`. (I _think_ that's the right thing to do here...)